### ISSUE TITLE
Crash on Ubuntu 15.04 with Python 3.4

### ISSUE THREAD

**Issue by graywizardx** [peterbrittain/asciimatics#12]
I was just running the samples and this fails on Ubuntu 15.04 with Python 3.4.

It appears there are two problems (both in screen.py):
* The stdout.write calls are receiving byte buffers, but stdout.write is expecting a string
* def print_at around L945 is expecting integer values, but on my console it is (somehow) receiving floats.

I am using guake, and zsh for the terminal, though I had the same issue in gnome terminal as well.

I attempted to patch, and got it partially working by adding .decode('utf-8') calls to all the contents of stdout.write (i.e. stdout.write(some_content.decode('utf-8')) ) but there were still some oddities in images.py with trailing escape sequences getting converted to strings instead of control characters.

Unfortunately I dont have time to look into this, so just wanted to log a quick issue in case it was something straight forward to correct.

**Comment by peterbrittain** [comment]
@graywizardx Looks like Some issues I had when I forest moved to Python 3 - e.g. int/float problems are usually because the calling function has done some maths that returns an int in Python 2 but a float in Python 3.  Can you please provide a stack or confirm which demo and which version of asciimatics you're trying to run?

**Comment by graywizardx** [comment]
I pulled your latest repo and am running that code ...

This is from just running 256colour.py:
```
Traceback (most recent call last):
  File "samples/256colour.py", line 22, in <module>
    Screen.wrapper(demo)
  File "/home/loki/projects/external/asciimatics/asciimatics/screen.py", line 479, in wrapper
    curses.wrapper(_wrapper)
  File "/usr/lib/python3.4/curses/__init__.py", line 94, in wrapper
    return func(stdscr, *args, **kwds)
  File "/home/loki/projects/external/asciimatics/asciimatics/screen.py", line 477, in _wrapper
    func(cur_screen)
  File "samples/256colour.py", line 18, in demo
    screen.play([Scene(effects, -1)], stop_on_resize=True)
  File "/home/loki/projects/external/asciimatics/asciimatics/screen.py", line 672, in play
    self.clear()
  File "/home/loki/projects/external/asciimatics/asciimatics/screen.py", line 879, in clear
    self._change_colours(self.COLOUR_WHITE, 0, 0)
  File "/home/loki/projects/external/asciimatics/asciimatics/screen.py", line 1445, in _change_colours
    sys.stdout.write(self._a_normal)
TypeError: must be str, not bytes
```

**Comment by peterbrittain** [comment]
@graywizardx Thanks for that quick response.

I know all the generic code runs on Python 3 as that is my standard test environment on Windows, but this stack shows that I have some compatibility issues from the move to terminfo.  It looks like tigetstr is not returning a suitable object to write to stdout.  I think decoding the bytes returned from those calls into strings at initialization will fix the bug instead of doing the decode on any write.   That said, I'll probably see fewer bug reports if I make the print logic more resilient to unexpected types. Will look into that when I have some time to set that up next week.

In the meantime, is it easy for you to get the stack trace for the division error too so I can look into that at the same time?

**Comment by peterbrittain** [comment]
Anyone else hitting a similar problem should try version 1.3.0 while waiting for the fix.  This is the last version that predates the move to terminfo.

**Comment by graywizardx** [comment]
I can look to get one. Can you tell me the L #s to change for the init decoding? If I get some time I can try and test it and give you the results.

**Comment by graywizardx** [comment]
Seems I cant reproduce the int error I was getting, I'll keep an eye out and see if I can figure out what happened last time.

**Comment by peterbrittain** [comment]
@graywizardx Appreciate your help here.  The line numbers are 1346-1353.  These calls to tigetstr should be decoded as per your original patch.  You should then no longer need the stdout decoding anymore.

Just a thought on the int problem.  Could it be the new fireworks demo?  I've not tested it on all platforms yet as it will be new for v1.5.0.

**Comment by graywizardx** [comment]
I'll see if I can get to it this weekend, have some project work at work to get out first.

**Comment by peterbrittain** [comment]
Thanks.  I'm similarly pushed for time this week...

**Comment by peterbrittain** [comment]
Fixed in v1.4.2 and v1.5.0.  New v1.4.2 download is now available for download from PYPI.
